sql/wsrep_thd.cc

```diff
@@ -346,11 +346,25 @@ bool wsrep_bf_abort(const THD* bf_thd, THD* victim_thd)
   WSREP_LOG_THD(victim_thd, "victim before");
   wsrep::seqno bf_seqno(bf_thd->wsrep_trx().ws_meta().seqno());
 
+  mysql_mutex_lock(&victim_thd->LOCK_thd_data);
   if (WSREP(victim_thd) && !victim_thd->wsrep_trx().active())
   {
-    WSREP_DEBUG("wsrep_bf_abort, BF abort for non active transaction");
-    wsrep_start_transaction(victim_thd, victim_thd->wsrep_next_trx_id());
+     WSREP_DEBUG("wsrep_bf_abort, BF abort for non active transaction");
+     switch (victim_thd->wsrep_trx().state()) {
```

Comment:
Why is the indentation changed to one extra space? It should be 2 spaces per indentation level.